ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts

```diff
@@ -1327,78 +983,388 @@ const triggerSet: TriggerSet<Data> = {
+        if (!num) {
+          console.error(`DFG Tower 1 Reminder: missing number: ${JSON.stringify(data.diveFromGraceNum)}`);
+          return;
+        }
+
+        const inout = output[data.diveFromGraceLashGnashKey]!();
+
+        if (data.eyeOfTheTyrantCounter === 1) {
+          // Tower 1 soaks only based on debuffs.
+          if (num === 3) {
+            if (data.diveFromGraceDir[data.me] === 'circle') {
+              if (data.diveFromGraceHasArrow[3])
+                return output.southTower1!({ inout: inout });
+              return output.circleTowers1!({ inout: inout });
+            } else if (data.diveFromGraceDir[data.me] === 'up') {
+              return output.upArrowTower1!({ inout: inout });
+            } else if (data.diveFromGraceDir[data.me] === 'down') {
+              return output.downArrowTower1!({ inout: inout });
+            }
+            return output.unknownTower!({ inout: inout });
+          }
+        } else if (data.eyeOfTheTyrantCounter === 2) {
+          // Tower 3 soaks based on debuffs and previous positions.
+          const pos = data.diveFromGracePreviousPosition[data.me];
+          if (num === 1) {
+            if (pos === 'middle')
+              return output.southTower3!({ inout: inout });
+            // Already soaked the number two towers.
+            if (pos === 'east' || pos === 'west')
+              return;
+          } else if (num === 2) {
+            if (pos === 'west')
+              return output.westTower3!({ inout: inout });
+            if (pos === 'east')
+              return output.eastTower3!({ inout: inout });
+          } else if (num === 3) {
+            return;
+          }
+          return output.unknownTower!({ inout: inout });
         }
-        // Outputs nothing if did not find debuff where there was an arrow
       },
       outputStrings: {
-        circlesDive: {
-          en: 'Dive (all circles)',
+        unknown: Outputs.unknown,
+        in: Outputs.in,
+        out: Outputs.out,
+        unknownTower: {
+          en: 'Tower (${inout})',
+        },
+        southTower1: {
+          en: 'South Tower (${inout})',
+        },
+        southTower3: {
+          en: 'South Tower (${inout})',
+        },
+        circleTowers1: {
+          en: 'Towers (circles, ${inout})',
+        },
+        circleTowers3: {
+          en: 'Towers (circles, ${inout})',
+        },
+        upArrowTower1: {
+          en: 'Up Arrow Tower (${inout})',
         },
-        southDive: {
-          en: 'South Dive',
+        downArrowTower1: {
+          en: 'Down Arrow Tower (${inout})',
         },
-        circlesDive2: {
-          en: 'Dive (all circles)',
+        upArrowTower3: {
+          en: 'Up Arrow Tower (${inout})',
         },
-        upArrowDive: {
-          en: 'Up Arrow Dive',
+        downArrowTower3: {
+          en: 'Down Arrow Tower (${inout})',
+        },
+        westTower3: {
+          en: 'West Tower (${inout})',
```

Comment:
Do these have a way to be like:
```suggestion
          en: Outputs.west.en + ' Tower (${inout})',
```

Reply:
Unfortunately, I don't think this helps anything, because everything still needs a translation.